src/screens/Login/index.tsx

```diff
@@ -39,9 +42,11 @@ const LoginScreen = () => {
     setFormSubmissionErrors('');
 
     try {
-      await AuthAdapter.login({ email, password });
+      const response = await AuthAdapter.login({ email, password });
+      const tokensResponse = response.data.attributes;
 
-      navigate('/');
+      setTokens(tokensResponse);
+      fetchUserProfile();
```

Comment:
Should we fetch the user profile every time when the user tries to access the protected route instead? To ensure that the user session is still valid before accessing the protected route. 🤔

Reply:
Yeah, that makes sense 👍 I updated here - 72dd143